src/run/run.ts

```diff
@@ -67,51 +68,83 @@ async function reportToTestrail(
     ? await extractTestResults(testRunConfig.projectId, testRunConfig.suiteId)
     : await extractTestResults();
 
-  if (isEmpty(results)) {
+  if (!results.length) {
     setFailed("No results for reporting to TestRail were found.");
     throw new Error();
   }
 
-  const { body: suite } = await testrailService.getTestSuite();
+  // Ensure there are no duplicate case ids
+  const case_ids = results.map((result) => result.case_id as number); // TODO: Stronger typing for results
+  const duplicate_case_ids = findDuplicates(case_ids);
 
-  if (isEmpty(suite)) {
-    setFailed("A TestRail Suite could not be found for the provided suite id.");
+  if (duplicate_case_ids.length) {
+    setFailed(
+      `Duplicate case ids found in test results: ${duplicate_case_ids.join(
+        ", "
+      )}. Please ensure that each test case is only reported once.`
+    );
     throw new Error();
   }
 
-  const milestone = await testrailService.establishMilestone();
+  const { body: suite } = await testrailService.getTestSuite().catch((error) => {
+    setFailed("A TestRail Suite could not be found for the provided suite id.");
+    throw error;
+  });
+
+  const milestone = await testrailService.establishMilestone().catch((error) => {
+    setFailed("A TestRail Milestone could not be established.");
+    throw error;
+  });
+
+  const datetime = new Intl.DateTimeFormat("en-CA", {
+    year: "numeric",
+    month: "2-digit",
+    day: "2-digit",
+    hour: "2-digit",
+    minute: "2-digit",
+    second: "2-digit",
+    timeZone: "America/Toronto",
+  }).format(new Date());
```

Comment:
Formatting is a little different, but not breaking. Is suffixed with `p.m.` instead of `PM`.

Reply:
I think that will be perfectly fine! :)